src/backend/parser/parse_type.c

```diff
@@ -360,6 +366,18 @@ typenameTypeMod(ParseState *pstate, const TypeName *typeName, Type typ)
 				 parser_errposition(pstate, typeName->location)));
 
 	typmodin = ((Form_pg_type) GETSTRUCT(typ))->typmodin;
+	typeoid = ((Form_pg_type) GETSTRUCT(typ))->oid;
+	typbasetype = ((Form_pg_type) GETSTRUCT(typ))->typbasetype;
+
+	if (dump_restore && (strcmp(dump_restore, "on") == 0) && !typmodin && typbasetype)
+	{
+		basetypeid = getBaseType(typeoid);
```

Comment:
Why call `getBaseType` when we already have it in `typbasetype`? The function `getBaseType` essentially returns the same thing.